src/extension.ts

```diff
@@ -31,20 +30,27 @@ export async function activate(context: vscode.ExtensionContext):
         // stop/start cycle in this situation is pointless, and doesn't work
         // anyways because the client can't be stop()-ped when it's still in the
         // Starting state).
-        if (clangdContext.clientIsStarting()) {
+        if (clangdContext && clangdContext.clientIsStarting()) {
           return;
         }
-        await clangdContext.dispose();
-        await clangdContext.activate(context.globalStoragePath, outputChannel);
+        if (clangdContext)
+          await clangdContext.dispose();
+        clangdContext =
+            await createContext(context.globalStoragePath, outputChannel);
+        if (clangdContext)
+          context.subscriptions.push(clangdContext);
         if (apiInstance) {
-          apiInstance.client = clangdContext.client;
+          apiInstance.client = clangdContext?.client;
         }
       }));
 
   let shouldCheck = false;
 
   if (vscode.workspace.getConfiguration('clangd').get<boolean>('enable')) {
-    await clangdContext.activate(context.globalStoragePath, outputChannel);
+    clangdContext =
+        await createContext(context.globalStoragePath, outputChannel);
```

Comment:
With a static factory method, this would then be:
```suggestion
        await ClangdContext.createContext(context.globalStoragePath, outputChannel);
```